camel/utils/commons.py

```diff
@@ -71,6 +71,31 @@ def wrapper(self, *args, **kwargs):
     return cast(F, wrapper)
 
 
+def azure_openai_api_key_required(func: F) -> F:
+    r"""Decorator that checks if the Azure OpenAI API key is available in the
+    environment variables.
+
+    Args:
+        func (callable): The function to be wrapped.
+
+    Returns:
+        callable: The decorated function.
+
+    Raises:
+        ValueError: If the Azure OpenAI API key is not found in the environment
+            variables.
+    """
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        if 'AZURE_OPENAI_API_KEY' in os.environ:
+            return func(self, *args, **kwargs)
+        else:
+            raise ValueError('Azure OpenAI API key not found.')
+
+    return cast(F, wrapper)
+
+
```

Comment:
for azure openai, I think not only api is required, the endpoint and api version also need to be checked

Reply:
I don’t think we need to check these variables here since they will be checked when initializing the AzureOpenAIModel instance. When running the commands, we should only check the AZURE_OPENAI_API_KEY

Reply:
`AZURE_OPENAI_API_KEY `, `AZURE_ENDPOINT `,  `AZURE_DEPLOYMENT_NAME `,  `AZURE_ API_VERSION` are all required by Azure, if we add `AZURE_OPENAI_API_KEY ` as decorator checking item, why not follow the same logic add others? Can we move the checking in initializing to this decorator? If not why we not also also check `AZURE_OPENAI_API_KEY`when initializing the AzureOpenAIModel instance? WDYT?